pandas/core/aggregation.py

```diff
@@ -5,12 +5,107 @@
 
 from collections import defaultdict
 from functools import partial
-from typing import Any, Callable, DefaultDict, List, Sequence, Tuple, Union
+from typing import (
+    Any,
+    Callable,
+    DefaultDict,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+from pandas._typing import Label
 
 from pandas.core.dtypes.common import is_dict_like, is_list_like
 
+from pandas.core.base import SpecificationError
 import pandas.core.common as com
 from pandas.core.indexes.api import Index
+from pandas.core.series import FrameOrSeriesUnion, Series
+
+
+def reconstruct_func(
+    func: Optional[
+        Union[
+            Union[Callable, str],
+            List[Union[Callable, str]],
```

Comment:
crazy signatures, can you create an alias to make this easier to read, *maybe* some of this is Label?